apps/chat/src/components/Marketplace/Marketplace.tsx

```diff
@@ -28,11 +34,31 @@ const Marketplace = () => {
   const isModelsLoading = useAppSelector(ModelsSelectors.selectModelsIsLoading);
   const models = useAppSelector(ModelsSelectors.selectModels);
 
-  const showOverlay = (isFilterbarOpen || isProfileOpen) && isSmallScreen();
-
   const [detailsModel, setDetailsModel] = useState<DialAIEntityModel>();
+  const [publishModel, setPublishModel] = useState<{
+    entity: DialAIEntityModel;
+    action: PublishActions;
+  }>();
   const [isMobile, setIsMobile] = useState(isSmallScreen());
 
+  const handleSetPublishEntity = useCallback(
+    (entity: DialAIEntityModel, action: PublishActions) =>
+      setPublishModel({ entity, action }),
+    [],
+  );
+
+  const handlePublishClose = useCallback(() => setPublishModel(undefined), []);
+
+  const showOverlay = (isFilterbarOpen || isProfileOpen) && isSmallScreen();
+
+  const entityForPublish: ShareEntity | null = publishModel?.entity
```

Comment:
maybe we can set publish model inside `handleSetPublishEntity` and get rid of 54-60 lines?